packages/react-components/react-combobox/src/components/Listbox/useListbox.ts

```diff
@@ -90,6 +89,27 @@ export const useListbox_unstable = (props: ListboxProps, ref: React.Ref<HTMLElem
     }
   };
 
+  const onFocus = (_event: React.FocusEvent<HTMLElement>) => {
+    // set initial active descendent upon focus, in the absence of a parent context controlling the state
+    if (!hasParentActiveDescendantContext) {
+      const activeDescendent = activeDescendantController.active();
+      if (activeDescendent) {
+        activeDescendantController.focus(activeDescendent);
+        return;
+      } else {
```

Comment:
another extremely minor nit is that you no longer need the `else`, since the earlier `if` has a `return`.

Reply:
You are so right!

Reply:
Fixed :)